train.py

```diff
@@ -0,0 +1,195 @@
+            transforms.Resize(256),
+            transforms.RandomHorizontalFlip(),
+            transforms.RandomVerticalFlip(),
+            transforms.ToTensor(),
+            # transforms.Normalize([0.4352, 0.5103, 0.2836], [0.2193, 0.2073,
+            # 0.2047])]),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])]),
+        target_transform=torch.tensor)
+
+    test_ds = AllCropsDataset(
+        data_path,
+        subset='test',
+        transform=transforms.Compose([
+            transforms.Resize(256),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])]),
+        target_transform=torch.tensor)
+
+    # print statistics
+    print('Train size:', len(train_ds))
+
+    print('Test size:', len(test_ds))
+
+    print('Number of samples in the dataset:', len(train_ds))
+
+    print('Crops in the dataset:', train_ds.crops)
+
+    print('Total number of classes in the dataset:', len(train_ds.classes))
+
+    print('Classes with the corresponding targets:')
+
+    print(train_ds.class_to_idx)
+
+    return train_ds, test_ds
+
+
+def split_on_train_and_test(random_seed, data_path, test_size):
+    for crop in os.listdir(data_path):
+        crop_path = os.path.join(data_path, crop)
+        _ = datadir_train_test_split(crop_path,
+                                     test_size=test_size,
+                                     random_state=random_seed)
+
+
+def main():
+
+    config = load_config('config/train_parametrs.yaml')
+
+    fix_random_seed(config['random_seed'], config['cudn_determenistic'])
+
+    print("Extract data")
+    unzip_data(config['data_zip_path'], config['data_path'])
+
+    print("Split on train and test")
+    split_on_train_and_test(
+        config['random_seed'],
+        config['data_path'],
+        config['test_size'])
+
+    print("Create datasets")
+    train_ds, test_ds = prepare_datasets(config['data_path'])
+
+    print("Create data loaders")
+    def train_set_d(index): return train_ds[index][0].float().numpy()
+    def test_set_d(index): return test_ds[index][0].float().numpy()
+    tri_train_set = TripletDataset(
+        torch.FloatTensor(
+            train_ds.targets).numpy(),
+        train_set_d,
+        len(train_ds),
+        config['n_sample'])
+    tri_test_set = TripletDataset(
+        torch.FloatTensor(
+            test_ds.targets),
+        test_set_d,
+        len(test_ds),
+        1)
+    tri_train_load = DataLoader(tri_train_set,
+                                batch_size=config['bath_size'],
+                                shuffle=True,
+                                num_workers=2,
+                                pin_memory=True
+                                )
+    tri_test_load = DataLoader(tri_test_set,
+                               batch_size=config['bath_size'],
+                               shuffle=False,
+                               num_workers=2,
```

Comment:
- 'tri' in the names 
- code duplication. Replace with the loop for ['train', 'test']